tests/conftest.py

```diff
@@ -1,80 +1,111 @@
 import os
-import re
-import io
 import warnings
 import pytest
 import uuid
-import wget
 import shutil
 import pathlib
-import hashlib
-import yaml
-import subprocess
-import requests
-from bs4 import BeautifulSoup
 
 import numpy as np
 
 import sqlalchemy as sa
 
-from astropy.time import Time
-from astropy.io import fits, votable
-from astropy.wcs import WCS
-
 from util.config import Config
-from models.base import FileOnDiskMixin, SmartSession, CODE_ROOT, _logger
+from models.base import FileOnDiskMixin, SmartSession, CODE_ROOT, get_all_database_objects, _logger
 from models.provenance import CodeVersion, Provenance
+from models.catalog_excerpt import CatalogExcerpt
 from models.exposure import Exposure
-from models.image import Image
-from models.datafile import DataFile
-from models.references import Reference
-from models.instrument import Instrument, get_instrument_instance
-from models.decam import DECam
-from models.source_list import SourceList
-from models.psf import PSF
-from models.world_coordinates import WorldCoordinates
-from models.zero_point import ZeroPoint
-from pipeline.data_store import DataStore
-from pipeline.preprocessing import Preprocessor
-from pipeline.detection import Detector
-from pipeline.astro_cal import AstroCalibrator
-from pipeline.photo_cal import PhotCalibrator
-from pipeline.catalog_tools import fetch_GaiaDR3_excerpt
-
-from util import config
+
 from util.archive import Archive
-from util.retrydownload import retry_download
-from util.exceptions import SubprocessFailure
+
+pytest_plugins = [
+    'tests.fixtures.simulated',
+    'tests.fixtures.decam',
+    'tests.fixtures.ztf',
+    'tests.fixtures.ptf',
+    'tests.fixtures.pipeline_objects',
+]
 
 
-# idea taken from: https://shay-palachy.medium.com/temp-environment-variables-for-pytest-7253230bd777
 # this fixture should be the first thing loaded by the test suite
-@pytest.fixture(scope="session", autouse=True)
-def tests_setup_and_teardown():
+# (session is the pytest session, not the SQLAlchemy session)
+def pytest_sessionstart(session):
     # Will be executed before the first test
     # print('Initial setup fixture loaded! ')
 
     # make sure to load the test config
     test_config_file = str((pathlib.Path(__file__).parent.parent / 'tests' / 'seechange_config_test.yaml').resolve())
-
     Config.get(configfile=test_config_file, setdefault=True)
+    FileOnDiskMixin.configure_paths()
 
-    yield
-    # Will be executed after the last test
-    # print('Final teardown fixture executed! ')
 
+# This will be executed after the last test (session is the pytest session, not the SQLAlchemy session)
+def pytest_sessionfinish(session, exitstatus):
+    # print('Final teardown fixture executed! ')
     with SmartSession() as session:
```

Comment:
For clarity, might be worth naming the parameter pytest_session, rather than naming it session and then overwriting that value with something that's a completely different type.

(For instance, if in some future we come back and need to do something with the pytest session, looking back at the function definition might give us the idea that "session" is the variable to use for that.)

Reply:
I did that, but pytest is really picky about the names of variables (think about fixtures as parameter names). 
I can call our session dbsession or something like that.